client/my-sites/stats/hooks/use-available-upgrade-tiers.tsx

```diff
@@ -107,43 +115,22 @@ function extendTiersBeyondHighestTier(
 				( highestTier?.views ?? 0 ) +
 				( highestTier.transform_quantity_divide_by ?? 0 ) * extendedTierCount;
 
-			const extendedTierAmount = extendedTierCount - purchasedExtendedTierCount;
-
 			availableTiers.push( {
 				minimum_price: totalPrice,
-				// Upgrade price for every 1M views.
-				upgrade_price: ( highestTier.per_unit_fee || 0 ) * extendedTierAmount,
+				upgrade_price: totalPrice - usageData?.current_tier?.minimum_price ?? 0,
```

Comment:
The `usageData?current_tier.minimum_price` would be limited to the 2M tier price when users have purchased an extended tier higher than 2M. We should calculate it with `usageData?.views_limit`.

Reply:
Should be addressed in https://github.com/Automattic/wp-calypso/pull/89429/commits/d5b8901d1814acfdf8f42761cce3da829f3d42e8.

Reply:
Added test for it: d27851d9688